modules/claims_api/app/controllers/claims_api/v2/veterans/evidence_waiver_controller.rb

```diff
@@ -75,6 +76,21 @@ def find_bgs_claim!(claim_id:)
 
           raise
         end
+
+        def file_number_check
+          @file_number = local_bgs_service.find_by_ssn(target_veteran.ssn)&.dig(:file_nbr) # rubocop:disable Rails/DynamicFindBy
+
+          if @file_number.nil?
+            raise ::Common::Exceptions::FailedDependency.new(
+              detail: "A BGS failure occurred while trying to retrieve Veteran 'FileNumber'"
+            )
+          end
+        rescue ::Common::Exceptions::FailedDependency => e
+          log_exception_to_sentry(e, nil, { message: e.errors[0].detail }, 'warn')
+          raise ::Common::Exceptions::ResourceNotFound.new(detail:
+            "Unable to locate Veteran's File Number. " \
+            'Please submit an issue at ask.va.gov or call 1-800-MyVA411 (800-698-2411) for assistance.')
+        end
```

Comment:
I think these are backwards. If FileNumber is nil and we didn't hit the rescue we should put the 422 with the File Number error... means we got a response but no FN..... then remove anything from the rescue type and just `rescue => e` at the bottom for troubleshooting. I think that's the intent for now. we can handle errors as we uncover them.